### ISSUE TITLE
Warnings with gcc 9.2 in Trilinos snapshot comparing span()

### ISSUE THREAD

**Issue by rppawlo** [kokkos/kokkos#2273]
In the kokkos snapshot in trilinos, when building with with gcc 9.2, getting the following warnings:
```
/ascldap/users/rppawlo/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:1261:19: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
 1261 |     if(dst.span() < std::numeric_limits<int>::max()) {
/ascldap/users/rppawlo/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:1293:17: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
 1293 |   if(dst.span() > std::numeric_limits<int>::max()) {
```
The line numbers are different in current kokkos repo but it looks like the offending code is still present.

**Comment by vqd8a** [comment]
Fixed in #2274

**Comment by rppawlo** [comment]
Thanks @vqd8a !
